services/web-app/utils/use-query-state.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright IBM Corp. 2021, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useMemo } from 'react'
+
+export const queryTypes = {
+  string: {
+    parse: (v) => v,
+    serialize: (v) => `${v}`
+  },
+  integer: {
+    parse: (v) => parseInt(v),
+    serialize: (v) => Math.round(v).toFixed()
+  },
+  float: {
+    parse: (v) => parseFloat(v),
+    serialize: (v) => v.toString()
+  },
+  boolean: {
+    parse: (v) => v === 'true',
+    serialize: (v) => (v ? 'true' : 'false')
+  },
+  timestamp: {
+    parse: (v) => new Date(parseInt(v)),
+    serialize: (v) => v.valueOf().toString()
+  },
+  isoDateTime: {
+    parse: (v) => new Date(v),
+    serialize: (v) => v.toISOString()
+  }
+}
+
+export const useQueryState = (
+  key,
+  {
+    defaultValue = '',
+    parse = queryTypes.string.parse,
+    serialize = queryTypes.string.serialize,
+    useStorage = false
+  }
+) => {
+  const router = useRouter()
+
+  // When not in a SSR context, get the query value directly from URLSearchParams so there's no
+  // reliance on router.query which may not be populated yet.
+  const getValue = useCallback(() => {
+    if (typeof window === 'undefined') {
+      return null
+    }
+
+    const query = new URLSearchParams(window.location.search)
+    const queryValue = query.get(key)
+    const storageValue = useStorage ? localStorage.getItem(`${router.pathname}:${key}`) : null
```

Comment:
Future us TODO:
Based on @jeanservaas decision on 11/2, filters should persist across a session, but when the session ends/a new one starts, the filters should all reset to the global defaults or team defaults.

Reply:
Okay, we can look into this later. I'm persisting sort and view, not the filter yet. I'd say I'm persisting view preferences so far, not any filters which reduce what is shown in the catalogs.